build_tools/aws-sdk-code-generator/templates/async_client_class.mustache

```diff
@@ -89,7 +89,11 @@ module {{module_name}}
         client: self,
         params: params,
         http_response: Seahorse::Client::Http::AsyncResponse.new,
-        config: config)
+        config: config,
+        tracer: config.telemetry_provider.tracer_provider.tracer(
```

Comment:
Config is already piped through. Couldn't request context have a method for tracer that just returned this thing?

Reply:
We could but I think this is simpler than creating a method and memorizing it.

Reply:
What do you mean? I'm suggesting a "tracer" method in RequestContext that looks at config for the tracer.

Reply:
The tracer isn't on the config though right?  We just use the tracer provider to create a new tracer for this request and Its scoped to a request and should only be created once per request I think?

Reply:
But a request context is made each request already right?

Reply:
Per our discussion, we are keeping this as it is.